FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_Sockets.c

```diff
@@ -2480,9 +2487,10 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t *pxSocket )
 
 		/* Check if the socket is valid, has type TCP and if it is bound to a
 		port. */
-		if( prvValidSocket( pxSocket, FREERTOS_IPPROTO_TCP, pdTRUE ) == pdFALSE )
+		if( ( prvValidSocket( pxSocket, FREERTOS_IPPROTO_TCP, pdTRUE ) == pdFALSE ) || 
+			( pvBuffer == NULL ) )
 		{
-			xByteCount = -pdFREERTOS_ERRNO_EINVAL;
+			xByteCount = FREERTOS_EINVAL;
```

Comment:
There is one situation in which it is valid to set `pvBuffer` to `NULL`: after the data have been read with the flag `FREERTOS_ZERO_COPY`.
Checking `prvValidSocket()` here should be enough.